libs/metrics/in_out.py

```diff
@@ -1,61 +1,263 @@
+                    paths[track_id] = sub_path
+                else:
+                    paths[track_id].extend(sub_path)
+        for track_id, data in paths.items():
+            path = data["path"]
+            new_in, new_out = cls.process_path(boundary_line, path)
+            people_in += new_in
+            people_out += new_out
+        return [people_in, people_out]
 
     @classmethod
     def get_trend_live_values(cls, live_report_paths: Iterator[str]) -> Iterator[int]:
-        raise NotImplementedError("Operation not implemented")
-
-    @classmethod
-    def get_weekly_report(cls, entities: List[str], number_of_weeks: int = 0,
-                          from_date: date = None, to_date: date = None) -> Dict:
-        raise NotImplementedError("Operation not implemented")
+        latest_in_out_results = {}
+        for n in range(10):
+            latest_in_out_results[n] = None
+        for live_path in live_report_paths:
+            with open(live_path, "r") as live_file:
+                lastest_10_entries = deque(csv.DictReader(live_file), 10)
+                for index, item in enumerate(lastest_10_entries):
+                    if not latest_in_out_results[index]:
+                        latest_in_out_results[index] = 0
+                    latest_in_out_results[index] += int(item["In"]) + int(item["Out"])
+        return [item for item in latest_in_out_results.values() if item is not None]
 
     @classmethod
     def get_in_out_file_path(cls, camera_id, config):
         """ Returns the path to the roi_contour file """
         return f"{get_source_config_directory(config)}/{camera_id}/{IN_OUT}/{IN_OUT}.json"
 
     @classmethod
-    def get_in_out_boundaries(cls, in_out_file_path):
+    def retrieve_in_out_boundary(cls, config, camera_id):
+        boundary_path = cls.get_in_out_file_path(camera_id, config)
+        boundary_line = cls.read_in_out_boundaries(boundary_path)
+        if boundary_line is None:
+            raise Exception(f"Camera {camera_id} does not have a defined in/out boundary")
+        else:
+            return boundary_line["in_out_boundary"]
+
+    @classmethod
+    def read_in_out_boundaries(cls, in_out_file_path):
         """ Given the path to the in-out file it loads it and returns it """
         if validate_file_exists_and_is_not_empty(in_out_file_path):
             with open(in_out_file_path) as json_file:
                 in_out_boundaries = json.load(json_file)
             return in_out_boundaries
         else:
             return None
+
+    @classmethod
+    def can_execute(cls, config, entity):
+        boundary_line = cls.read_in_out_boundaries(cls.get_in_out_file_path(entity["id"], config))
+        if boundary_line is None:
+            return False
+        return True
+
+# Auxiliary methods taken from:
+# https://github.com/yas-sim/object-tracking-line-crossing-area-intrusion/blob/master/object-detection-and-line-cross.py
+def check_line_cross(boundary_line, trajectory):
```

Comment:
Maybe we can move these functions to a `utils` file. What do you think?

Reply:
Hmm I thought about reducing complexity here, although it will be used only in in/out. I'd maybe leave it here until the next iteration (multiple lines, areas, etc)